pika/channel.py

```diff
@@ -1352,16 +1385,46 @@ def _on_unexpected_frame(self, frame_value):
         """
         LOGGER.error('Unexpected frame: %r', frame_value)
 
-    def _validate_channel_and_callback(self, callback):
-        """Verify that channel is open and callback is callable if not None
+    def _validate_channel(self):
+        """Verify that channel is open
 
         :raises ChannelClosed: if channel is closed
-        :raises ValueError: if callback is not None and is not callable
+
         """
         if not self.is_open:
             raise exceptions.ChannelClosed()
-        if callback is not None and not is_callable(callback):
-            raise ValueError('callback must be a function or method')
+
+    def _require_callback(self, callback):
+        """Require that callback is callable and is not None
+
+        :raises: ValueError
```

Comment:
By the way, would `TypeError` be more appropriate? I think it might be, because we're looking for callback to be of callable type?